dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java

```diff
@@ -70,7 +70,9 @@
     AGE( Date.class, false ),
     URL( String.class, false ),
     FILE_RESOURCE( String.class, false ),
-    IMAGE( String.class, false );
+    IMAGE( String.class, false ),
+    // NOT QUITE SURE HOW TO LINK THIS CLASS WITH THE COMPLEX and vv.
+    COMPLEX( ComplexValueType.class, false );
```

Comment:
Maybe we need another parameter to the constructor? some kind of additional option class..